ds-caselaw-ingester/lambda_function.py

```diff
@@ -315,6 +298,18 @@ def insert_document_xml(uri, xml) -> bool:
         return False
 
 
+def select_uri(source_uri):
+    """If the judgment exists, suggest a URL that is not going to be collided with,
+    even if there are multiple requests a second or the ingester is restarted."""
+    global collision_counter
```

Comment:
If there are multiple invocations of the lambda a second, are they run in the same runtime environment? I would imagine theyd be in different runtimes and so the global variable wouldnt be shared. I'd suggest that we could add a hash or a unique identifier from the lambda run /event message or something along those lines? TDR ref even?

Reply:
... that's a very valid point